File: frameworks/domain.core/src/test/java/com/everest/frameworks/domain/core/GlobalIdGeneratorTest.java
package com.everest.frameworks.domain.core;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class GlobalIdGeneratorTest {
    @Test
    void should_be_able_to_generate_identities_from_underlying_identity_service() {
        GlobalIdGeneratorResetter.reset(new UuidBasedIdGenerator());
        assertThat(GlobalIdGenerator.nextId()).hasSize(32);
        assertThat(GlobalIdGenerator.nextReadableId()).hasSize(21);
    }
}

File: frameworks/domain.core/src/test/kotlin/com/qomolangma/frameworks/domain/core/GlobalIdGeneratorTest.kt
package com.qomolangma.frameworks.domain.core

import com.qomolangma.frameworks.domain.core.GlobalIdGenerator.nextId
import com.qomolangma.frameworks.domain.core.GlobalIdGenerator.nextReadableId
import com.qomolangma.frameworks.domain.core.GlobalIdGeneratorResetter.reset
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class GlobalIdGeneratorTest {
    @Test
    fun should_be_able_to_generate_identities_from_underlying_identity_service() {
        reset(UuidBasedIdGenerator())
        assertThat(nextId()).hasSize(32)
        assertThat(nextReadableId()).hasSize(21)
    }
}
